blockprod/src/lib.rs

```diff
@@ -19,15 +19,15 @@ use std::sync::Arc;
 
 use chainstate::ChainstateHandle;
 use common::{
-    chain::{block::BlockCreationError, ChainConfig, Destination},
+    chain::{block::BlockCreationError, ChainConfig, GenBlock},
+    primitives::{BlockHeight, Id},
     time_getter::TimeGetter,
 };
 use consensus::ConsensusVerificationError;
-use detail::{builder::PerpetualBlockBuilder, BlockProduction};
-use interface::BlockProductionInterface;
+use detail::{BlockProduction, JobKey};
+use interface::blockprod_interface::BlockProductionInterface;
 use mempool::MempoolHandle;
 use subsystem::subsystem::CallError;
-use tokio::sync::mpsc;
 
 #[derive(thiserror::Error, Debug, PartialEq, Eq)]
 pub enum BlockProductionError {
```

Comment:
We can probably delete a lot of these as they're no longer being used

Reply:
I removed those that aren't used and kept ChainstateChannelClosed because we need it for NewTip

Reply:
Ah of course. Ok, this can be cleaned up later.